### ISSUE TITLE
ChromeOS: OS-level notification from inappropriate websites

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#938]
# Issue
On my son's Chromebook he keeps seeing OS-level notification from inappropriate websites that he accidentally accessed and likely pressed some luring buttons (he was googling science projects' material then he thought clicking those was necessary to move on seeing contents). Now, OS offers to turn off those notifications, but it's unclear if turning off the notification is enough. Those webpages are not explicitly opened on the Chrome web browser.

# CoS
- [x] Stopped seeing the notifications from the particular websites <s>withOUT turning off the notification from those websites.</s> --> https://github.com/kinu-garage/hut_10sqft/issues/938#issuecomment-1834582639
- [x] (Option but wanted) Understand how those web site that is NOT open on the web browser can send notifications.
- [x] (Option but strongly wanted) Implement prevention (my son can easily fall for those predator websites, and it's not necessarily his fault).

**Comment by 130s** [comment]
Needing an immediate resolution, I went with site-by-site approach. Following this [quora post](https://qr.ae/pKwyyZ).
> In your Google Chrome, go to Settings>Notifications>Sites. You'll have a list of all the sites that have notification access. Based on the notification content, you should be able to figure out the offending site and block notifications for it.

**Comment by 130s** [comment]
> - [ ] (Option but wanted) Understand how those web site that is NOT open on the web browser can send notifications.

I guess its web browser's Push Notification feature (explained in [stackoverflow.com#58454545](https://stackoverflow.com/questions/58454545/how-does-website-display-notifications-after-its-page-closed).

> - [ ]  (Option but strongly wanted) Implement prevention (my son can easily fall for those predator websites, and it's not necessarily his fault).

## Solution 
TLDR I implemented the following solution. IDK if this is the best, I hope to see more organized, systematic solution but haven't found such a thing so far. 
- manually disabled the notification on the Chrome web browser (`chrome://settings/content/notifications`) on his Chromebook.
   - On ChromeOS where applications are implemented as Chrome web browser (which I haven't verified though), other apps that are used in our usecases rely on the notifications, most notable `Google Chat`, may be affected. To avoid the possibility, I explicitly added those apps in the same page as "approved app".

Note If our accounts are under Google Workspace then admin feature might be available e.g. [reddit.com#k12sysadmin/skfe7a](https://www.reddit.com/r/k12sysadmin/comments/skfe7a/disable_allowed_notifications_on_a_chromebook/?onetap_auto=true).

### Verification of the solution
- [x] I verified apps that need to notify still show notification using G-Chat.
- [ ] I verified the notification from random website does NOT show notification.

**Comment by 130s** [comment]
> - [ ] (Option but wanted) Understand how those web site that is NOT open on the web browser can send notifications.

I guess its web browser's Push Notification feature (explained in [stackoverflow.com#58454545](https://stackoverflow.com/questions/58454545/how-does-website-display-notifications-after-its-page-closed).

> - [ ]  (Option but strongly wanted) Implement prevention (my son can easily fall for those predator websites, and it's not necessarily his fault).

## Solution 
TLDR I implemented the following solution. IDK if this is the best, I hope to see more organized, systematic solution but haven't found such a thing so far. 
- manually disabled the notification on the Chrome web browser (`chrome://settings/content/notifications`) on his Chromebook.
   - On ChromeOS where applications are implemented as Chrome web browser (which I haven't verified though), other apps that are used in our usecases rely on the notifications, most notable `Google Chat`, may be affected. To avoid the possibility, I explicitly added those apps in the same page as "approved app".

### Background of the suggested solution

- Limited to kids' Google account, the web pages the chrome browser on their account can access are already filtered via Google FamilyLink feature. That said, the push notifications that I witnessed this time on my son's Chromebook did show inappropriate content (I was horrified when I saw a thumbnail that includes nudity), which indicates that push notification may be out of bounce from FamilyLink's control. Given this limitation, I chose to disable the entire push notification on the chrome browser, risking regression on existing usecases as mentioned in the solution section.
- Note: If our accounts are under Google Workspace then admin feature might be available e.g. [reddit.com#k12sysadmin/skfe7a](https://www.reddit.com/r/k12sysadmin/comments/skfe7a/disable_allowed_notifications_on_a_chromebook/?onetap_auto=true).

### Verification of the solution
- [x] I verified apps that need to notify still show notification using G-Chat.
- [ ] I verified the notification from random website does NOT show notification.
